test/lib/commands/copy.js

```diff
@@ -0,0 +1,221 @@
+  assertExists(path.join('build', 'package.json'))
+  assertExists(path.join('build', 'README.md'))
+  assertExists(path.join('build', 'lib', 'index.js'))
+
+  // src should not be copied because it's excluded by files.
+  assertMissing(path.join('build', 'src'))
+})
+
+t.test('should copy dependencies', async t => {
+  const { npm, outputs, logs } = await load(t, {
+    prefixDir: {
+      'package.json': JSON.stringify({
+        name: 'test-package',
+        version: '1.0.0',
+        dependencies: {
+          foo: '^1.0.0',
+          bar: '^1.0.0',
+        },
+      }),
+      node_modules: {
+        foo: { },
+        bar: { },
+        baz: { },
+      },
+    },
+  })
+  process.chdir(npm.prefix)
+  await npm.exec('copy', ['build'])
+  t.strictSame(outputs, [])
+  t.strictSame(logs.notice, [])
+
+  assertExists(path.join('build', 'node_modules', 'foo'))
+  assertExists(path.join('build', 'node_modules', 'bar'))
+  // baz is missing because it is an extraneous dep.
+  assertMissing(path.join('build', 'node_modules', 'baz'))
+})
+
+t.test('should not copy bundled dependencies if they are omitted', async t => {
+  const { npm, outputs, logs } = await load(t, {
+    prefixDir: {
+      'package.json': JSON.stringify({
+        name: 'test-package',
+        version: '1.0.0',
+        bundledDependencies: true,
+        optionalDependencies: {
+          foo: '^1.0.0',
+        },
+      }),
+      node_modules: {
+        foo: {
+          'package.json': JSON.stringify({
+            name: 'foo',
+            version: '1.0.0',
+          }),
+        },
+      },
+    },
+  })
+  process.chdir(npm.prefix)
+  npm.config.set('omit', ['optional'])
+  await npm.exec('copy', ['build'])
+  t.strictSame(outputs, [])
+  t.strictSame(logs.notice, [])
+
+  assertMissing(path.join('build', 'node_modules', 'foo'))
+})
+
+t.test('workspaces', async t => {
+  const fixture = {
+    prefixDir: {
+      'package.json': JSON.stringify({
+        name: 'workspace-root',
+        workspaces: [
+          'pkgs/a',
+          'pkgs/b',
+        ],
+      }),
+      pkgs: {
+        a: {
+          'package.json': JSON.stringify({
+            name: 'a',
+            version: '1.0.0',
+          }),
+          'README.md': 'a',
+        },
+        b: {
+          'package.json': JSON.stringify({
+            name: 'b',
+            version: '1.0.0',
+            dependencies: { c: '^1.0.0' },
+          }),
+          'README.md': 'b',
+        },
+      },
+      node_modules: {
+        a: t.fixture('symlink', '../pkgs/a'),
+        b: t.fixture('symlink', '../pkgs/b'),
+        c: t.fixture('symlink', '../pkgs/a'),
+      },
+    },
+  }
+
+  t.test('should only copy included workspaces', async t => {
+    const { npm } = await load(t, fixture)
+    process.chdir(npm.prefix)
+    npm.config.set('workspace', ['a'])
+    await npm.exec('copy', ['build'])
+
+    assertExists(path.join('build', 'node_modules', 'a'))
```

Comment:
Maybe we should test for `package.json` and `README.md` copied, not just for the folder?